### PR TITLE
[pt2] add metas for `multilabel_margin_loss` ops

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #104388
* #104236
* #104167

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -377,6 +377,78 @@ def meta_multi_margin_loss_backward(
...
+
+
+@register_meta(aten.multilabel_margin_loss_forward)
+@out_wrapper("output", "is_target")
+def meta_multilabel_margin_loss_forward(
+    input: Tensor,
+    target: Tensor,
+    reduction: int,
+) -> Tuple[Tensor, Tensor]:
+    target_arg = "argument #2 'target'"
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:417]: I didn't port the helper `TensorArg` class (that's used in core) and inlined the error message instead.

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -377,6 +377,78 @@ def meta_multi_margin_loss_backward(
...
+    ndims = input.ndim
+    _multilabel_margin_loss_shape_check(ndims, target_arg, input, target)
+    torch._check(
+        target.shape == is_target.shape,
+        lambda: (
+            f"Expected tensor for {target_arg} to have same size as tensor for {is_target_arg}"
+            f"; but {target.shape} does not equal {is_target.shape}"
+            f" (while checking arguments for multilabel_margin_loss_backward)"
+        ),
+    )
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:448]: In core, this is called `checkSameSize`. I've just inlined it here. We already have `check_same_shape` in `prims_common`, but the error message there is different.

**Comment by nkaretnikov** [comment]
This was opened before Horace said on the multi_margin_loss PR that these should probably be defined as decomps.

Can this still be merged? I can add a decomp later.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**